src/js/angular/ttyg/controllers/agent-settings-modal.controller.js

```diff
@@ -1,42 +1,275 @@
+import {decodeHTML} from "../../../../app";
+import {ExtractionMethod} from "../../models/ttyg/agents";
+import 'angular/core/services/similarity.service';
+import 'angular/core/services/connectors.service';
+import 'angular/rest/repositories.rest.service';
+import {REPOSITORY_PARAMS} from "../../models/repository/repository";
+
 angular
-    .module('graphdb.framework.ttyg.controllers.agent-settings-modal', [])
+    .module('graphdb.framework.ttyg.controllers.agent-settings-modal', [
+        'graphdb.framework.core.services.similarity',
+        'graphdb.framework.core.services.connectors',
+        'graphdb.framework.rest.repositories.service'
+    ])
     .controller('AgentSettingsModalController', AgentSettingsModalController);
 
-AgentSettingsModalController.$inject = ['$scope', '$uibModalInstance', 'toastr', 'UriUtils', '$translate', 'dialogModel'];
+AgentSettingsModalController.$inject = ['$scope', '$uibModalInstance', 'SimilarityService', 'ConnectorsService', 'RepositoriesRestService', '$sce', 'toastr', 'UriUtils', '$translate', 'dialogModel'];
+
+function AgentSettingsModalController($scope, $uibModalInstance, SimilarityService, ConnectorsService, RepositoriesRestService, $sce, toastr, UriUtils, $translate, dialogModel) {
+
+    // =========================
+    // Private variables
+    // =========================
 
-function AgentSettingsModalController($scope, $uibModalInstance, toastr, UriUtils, $translate, dialogModel) {
+    const CHAT_GPT_RETRIEVAL_CONNECTOR_NAME = 'ChatGPT Retrieval';
 
     // =========================
     // Public variables
     // =========================
 
+    /**
+     * The model used in the form.
+     * @type {AgentFormModel|*}
+     */
+    $scope.agentFormModel = dialogModel.agentFormModel;
+
+    /**
+     * The active repository info model.
+     * @type {RepositoryInfoModel|*}
+     */
+    $scope.activeRepositoryInfo = dialogModel.activeRepositoryInfo;
+
+    /**
+     * Flags used to show/hide the loader for the extraction methods when the user opens a panel and some data needs to
+     * be loaded.
+     * @type {{[ExtractionMethod.SIMILARITY]: boolean}}
```

Comment:
It may need to be {[string]: boolean} with the description that the keys should be values from ExtractionMethod.

Reply:
fixed